src/daemon/absorber.cc

```diff
@@ -137,6 +157,22 @@ void Absorber::DoExecute(const Atomic<bool>& is_shutting_down) {
 
     Universal outgoing(new net::proto::Universal);
 
+    raii_scope_deleter asan_blacklist_deleter;
+    if (incoming->has_asan_blacklist() && incoming->flags().has_asan_blacklist_file()) {
+      g_asan_blacklist_file_write_lock.lock();
```

Comment:
If I understand this code correctly, then you block all absorber threads just to wait for a single temporary file creation. There is a much more simple solution, like using already implemented `base::TemporaryDir` per task execution. You should create this folder at the beginning, put the file with predefined name, e.g. "sanitize_blacklist.txt", and expect that more files may appeat in the flags in the future. All the stuff around this mutex and `TmpUniqFile` seems unnecessary.